wicket/src/ui/panes/update.rs

```diff
@@ -0,0 +1,286 @@
+            y: 0,
+        };
+        frame.render_widget(popup, full_screen);
+    }
+
+    fn update_items(&mut self, state: &State) {
+        let versions = state.update_state.artifact_versions.clone();
+
+        self.items = state
+            .update_state
+            .items
+            .iter()
+            .map(|(id, states)| {
+                let children: Vec<_> = states
+                    .iter()
+                    .map(|(artifact, s)| {
+                        let version = artifact_version(artifact, &versions);
+                        let spans = vec![
+                            Span::styled(
+                                artifact_title(*artifact),
+                                style::selected(),
+                            ),
+                            Span::styled("UNKNOWN", style::selected_line()),
+                            Span::styled(version, style::selected()),
+                            Span::styled(s.to_string(), s.style()),
+                        ];
+                        TreeItem::new_leaf(align_by(
+                            0,
+                            25,
+                            self.contents_rect,
+                            spans,
+                        ))
+                    })
+                    .collect();
+                TreeItem::new(*id, children)
+            })
+            .collect();
+    }
+}
+
+fn artifact_version(
+    artifact: &KnownArtifactKind,
+    versions: &BTreeMap<KnownArtifactKind, String>,
+) -> String {
+    versions.get(artifact).cloned().unwrap_or_else(|| "UNKNOWN".to_string())
+}
+
+impl Control for UpdatePane {
+    fn is_modal_active(&self) -> bool {
+        self.popup_open
+    }
+
+    fn resize(&mut self, state: &mut State, rect: Rect) {
+        self.rect = rect;
+        let chunks = Layout::default()
+            .direction(Direction::Vertical)
+            .constraints(
+                [
+                    Constraint::Length(3),
+                    Constraint::Length(3),
+                    Constraint::Min(0),
+                    Constraint::Length(3),
+                ]
+                .as_ref(),
+            )
+            .split(rect);
+        self.title_rect = chunks[0];
+        self.table_headers_rect = chunks[1];
+        self.contents_rect = chunks[2];
+        self.help_rect = chunks[3];
+
+        self.update_items(state);
+    }
+
+    fn on(&mut self, state: &mut State, event: Event) -> Option<Action> {
+        match event {
+            Event::Term(TermEvent::Key(e)) => match e.code {
+                KeyCode::Up => {
+                    // Keep the rack selection in sync across panes
+                    state.rack_state.prev();
+                    self.tree_state.key_up(&self.items);
+                    Some(Action::Redraw)
+                }
+                KeyCode::Down => {
+                    // Keep the rack selection in sync across panes
+                    state.rack_state.next();
```

Comment:
This doesn't actually work because the tree has more elements (its subcomponents) than the rack. We'd need to keep track manually to make it work, but I don't think it's worth worrying about right now, so I'm going to remove it.

Reply:
I fixed this in a follow up, with some judicious usage of the `TreeIdentifierVec` from the tui tree we are using. Huge thanks to @sunshowers for listening to me rant and pointing out something I should have recognized in the code.